optuna/samplers/_cmaes.py

```diff
@@ -288,6 +315,25 @@ def _log_independent_sampling(self, trial: FrozenTrial, param_name: str) -> None
             )
         )
 
+    def _get_trials(self, study: "optuna.Study") -> List[FrozenTrial]:
+        complete_trials = []
+        for t in study.get_trials(deepcopy=False):
+            if t.state == TrialState.COMPLETE:
+                complete_trials.append(t)
+            elif (
+                t.state == TrialState.PRUNED
+                and len(t.intermediate_values) > 0
+                and self._consider_pruned_trials
+            ):
+                _, value = max(t.intermediate_values.items())
+                if value is None:
+                    continue
+                # We rewrite the value of the trial `t` for sampling, so we need a deepcopy.
```

Comment:
In favor of performance, we could modify the code when the value is read (https://github.com/optuna/optuna/pull/1229/files#diff-1351672280ea468a02653b364cb473c8R232), rather than creating copies and setting the `value` attributes. What do you think? It might increase code complexity but it could be a TODO.

Reply:
Thanks for the comment! I agree with you that the current implementation is inefficient, but I'm not sure how to realize your suggestion. I would like to work on your suggestion in another PR.

Reply:
I think you can modify the code where this `value` is read, to extract the "value" from the intermediate values if the state is pruned.